.github/workflows/integration_test.yml

```diff
@@ -0,0 +1,259 @@
+    secrets:
+      extra-env:
+        required: false
+    inputs:
+      integration_tests_goal:
+        description: integration-tests goal
+        required: true
+        type: string
+      plugin_name:
+        description: The plugin name
+        required: true
+        type: string
+      ddl_file:
+        description: The DDL file for the plugin
+        required: false
+        default: ''
+        type: string
+      integration_tests_ref:
+        description: The ref of killbill-integration-tests to checkout
+        required: false
+        default: 'refs/heads/work-for-release-0.23.x'
+        type: string
+      maven_args:
+        description: The arguments to pass to all Maven commands
+        required: false
+        default: '-Dmaven.test.redirectTestOutputToFile=true -Dcheck.skip-dependency=true -Dcheck.skip-dependency-scope=true -Dcheck.skip-dependency-versions=true -Dcheck.skip-duplicate-finder=true -Dcheck.skip-enforcer=true -Dcheck.skip-rat=true -Dcheck.skip-spotbugs=true -DskipTests=true'
+        type: string
+      maven_goal:
+        description: The Maven goal(s) used to build the plugin
+        required: false
+        default: 'clean package'
+        type: string
+      maven_version:
+        description: The Maven version used for builds
+        required: false
+        default: '3.8.4'
+        type: string
+      max-parallel:
+        description: 'max parallel jobs'
+        required: false
+        default: 100
+        type: number
+      timeout-minutes:
+        description: 'timeout-minutes used by the builds (defaults to 360)'
+        required: false
+        default: 360
+        type: number
+      fail-fast:
+        description: Determinate if matrix should fail fast
+        required: false
+        default: true
+        type: boolean
+
+jobs:
+  integration_test:
+    name: jdk-${{ matrix.java-version }} ${{ matrix.database-adapter }}
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: ${{ inputs.fail-fast }}
+      matrix:
+        include:
+          - ruby-version: '2.4.2'
+            database-adapter: 'mysql2'
+            database-user: 'root'
+            database-password: 'root'
+            database-port: '3306'
+            docker-compose-file: 'docker-compose.ci.mysql.yml'
+            java-version: '11'
+            java-distribution: 'temurin'
+          - ruby-version: '2.4.2'
+            database-adapter: 'postgresql'
+            database-user: 'postgres'
+            database-password: 'postgres'
+            database-port: '5432'
+            docker-compose-file: 'docker-compose.ci.postgresql.yml'
+            java-version: '11'
+            java-distribution: 'temurin'
+      max-parallel: ${{ inputs.max-parallel }}
+    steps:
+      - name: Set environment variables
+        run: |
+          echo '${{ secrets.extra-env }}' | jq -r 'to_entries|map("::add-mask::\(.value|tostring)")|.[]'
```

Comment:
Just for my understanding: Are those a map of all our secrets on GH?

Reply:
They are selectively sent, e.g. https://github.com/killbill/killbill-avatax-plugin/blob/d3c90df4a660e81db98bd39ed0a8ad012c07452a/.github/workflows/ci.yml#L28